lib/mongodb.js

```diff
@@ -170,11 +171,32 @@ MongoDB.prototype.fromDatabase = function (model, data) {
         // Convert the Binary into String
         data[p] = data[p].toString();
       }
+    } else if(prop && prop.type === GeoPoint) {
+      data[p] = new GeoPoint({ lat: data[p].coordinates[1], lng: data[p].coordinates[0] })
```

Comment:
I think it should be ok to return a plain object here, that's what we are already doing in the postgresql connector:
 - https://github.com/strongloop/loopback-connector-postgresql/blob/8837e6d5e2b7210f3a286fb0f715fd7c1936dda8/lib/postgresql.js#L234-L242
 - https://github.com/strongloop/loopback-connector-postgresql/blob/8837e6d5e2b7210f3a286fb0f715fd7c1936dda8/lib/postgresql.js#L465-L470